src/appProvider.js

```diff
@@ -15,16 +15,13 @@ import { store, history } from './redux/store';
 
 import '../src/style/styles.less';
 
-const defaultidx = localStorage.getItem('localindex') || 0;
-const locales = [AppLocale.en, AppLocale.zh];
-class AppProvider extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { locale: defaultidx };
-    this.langHandler = this.langHandler.bind(this);
+export default class AppProvider extends Component {
+  locales = [AppLocale.en, AppLocale.zh]
+  state = {
+    locale: localStorage.getItem('localindex') || 0,
```

Comment:
is this a built-in state var? `localindex`?

Reply:
I'm not exactly sure why we're using this. My guess would be maybe this is used for electron.

Reply:
this is actually for our localization

Reply:
it's used to set the default language for the browser users before. Let them change language one time, and in the future, when they open our app, it will automatically show the language they choose last time. But it seems not work in electron?

Reply:
i think its fine for electron. havent seen any issues yet.